tests/unit/middlewares/withAuth.test.ts

```diff
@@ -0,0 +1,261 @@
+  return {
+    config: {
+      connectAccountSessionSalt: "#*b+x3ZXE3-h[E+Q5YC5`jr~y%CA~R-[",
+      connectOpenIdConfigurationUrl: "",
+      connectApplicationClientId: "",
+      connectApplicationClientSecret: "c9ab0fdc-b2dc-47ad-933b-87cf1b180ab5",
+      connectAccountRedirectUri: "",
+      connectAudience: "",
+      connectApplicationScopes: "",
+      connectProviderUrl: "http://foo.test",
+      connectAccountURL: "http://foo.test",
+      dynamoRegion: "eu-west-3",
+    },
+    oauth2Client: {
+      verifyJWT: async () => {
+        mockedVerifyJWTCall = +1;
+
+        return {
+          sub: "2a14bdd2-3628-4912-a76e-fd514b5c27a8",
+        };
+      },
+    },
+  };
+});
+
+async function sealJWS(access_token: string, salt?: string): Promise<string> {
+  return seal(
+    {
+      persistent: {
+        "user-session": {
+          access_token,
+          sub: "2a14bdd2-3628-4912-a76e-fd514b5c27a8",
+        },
+      },
+      flash: {},
+    },
+    salt || config.connectAccountSessionSalt,
+    defaults,
+  );
+}
+
+function mockedReqAndRes(
+  sealedJWE: string,
+): {
+  mockedExtendedRequest: ExtendedRequest;
+  mockedResponse: ServerResponse;
+} {
+  const mockedRequest = new IncomingMessage(new Socket());
+  const mockedResponse = new ServerResponse(mockedRequest);
+
+  const mockedSession = {
+    set: () => {
+      return;
+    },
+    get: () => {
+      return undefined;
+    },
+    unset: () => {
+      return;
+    },
+    destroy: () => {
+      return;
+    },
+    save: async () => {
+      return;
+    },
+  };
+
+  const mockedExtendedRequest = Object.assign(mockedRequest, {
+    session: mockedSession,
+    query: {},
+    cookies: {},
+    body: null,
+    env: {},
+  });
+
+  mockedExtendedRequest.headers = {
+    cookie: `connect-account-session=${sealedJWE}`,
+  };
+  mockedExtendedRequest.rawHeaders = [
+    "Cookie",
+    `connect-account-session=${sealedJWE}`,
+  ];
+
+  return { mockedExtendedRequest, mockedResponse };
+}
+
+const mockedHandler: Handler = async (
+  _mockedExtendedRequest,
+  _mockedResponse,
+) => {
+  return _mockedResponse.end();
+};
+
+describe("withAuth", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("should redirect to the login flow if no UserCookie is provided", async (done) => {
+    expect.assertions(3);
+
+    const access_token = generateHS256JWS();
+    const sealedJWS = await sealJWS(
+      access_token,
+      "thisIsAWrongSaltToMakeUserCookieUndefined",
+    );
+
+    const { mockedExtendedRequest, mockedResponse } = mockedReqAndRes(
+      sealedJWS,
+    );
+
+    const withAuthCallback = wrapMiddlewares(
+      [withSession, withAuth],
+      mockedHandler,
+    );
+    await withAuthCallback(mockedExtendedRequest, mockedResponse);
+
+    expect(mockedResponse.statusCode).toEqual(HttpStatus.TEMPORARY_REDIRECT);
+    expect(spiedOnDecryptVerifyAccessToken).not.toHaveBeenCalled();
+    expect(mockedVerifyJWTCall).toEqual(1);
```

Comment:
I'm not fond of this way of doing.
Did you try with something like `expect(verifyJWT).toHaveBeenCalledOnce()` (or smth like that)?

Reply:
There is no `spyOn` on `verifyJWT` because it is part of `config`, and we have to mock the file with `jest.mock`, and adding a `spyOn` would get overwritten by it.

Reply:
Ok if you can't.
It is mocked this way on sparta-api: https://github.com/fewlinesco/sparta-monorepo/blob/167d184c3fa8b726742d47dc46ed6738870652de/services/sparta-api/test/queries/program.test.ts#L11

Reply:
Thanks for the example, I'll have a look tomorrow 👍.